File: AtlasWeb/Core/wwwroot/app/NebulaClient/StorageManager/DefinitionManagement/TTObjectRelationSubtypeDef.ts
import { Guid } from "../../Mscorlib/Guid";
import { TTObjectDef } from "./TTObjectDef";
import { TTObjectRelationDef } from "./TTObjectRelationDef";

export class TTObjectRelationSubtypeDef {
    public RelationSubtypeDefID: Guid;
    public RelationDef: TTObjectRelationDef;
    public RelationDefID: Guid;
    public ChildObjectDef: TTObjectDef;
    public ChildObjectDefID: Guid;
    public ParentObjectDef: TTObjectDef;
    public ParentObjectDefID: Guid;
    public CodeName: string;
    public ParentName: string;
    public ChildrenName: string;
}

File: AtlasModel/Enterprise/TTObjectRelationSubtypeDef.cs
using System;

namespace AtlasModel.Enterprise
{
    public partial class TTObjectRelationSubtypeDef
    {
        public Guid RelationSubtypeDefID { get; set; }
        public Guid RelationDefID { get; set; }
        public Guid ParentObjectDefID { get; set; }
        public Guid ChildObjectDefID { get; set; }
        public string ParentName { get; set; }
        public string ChildrenName { get; set; }
        public short CheckOutStatus { get; set; }
    }
}